### DISCUSSION TITLE
Is it possible to move PNCs in game ?

### DISCUSSION THREAD

**Discussion by BuzzLEquerre** [flareteam/flare-game#976] [General]
Hello, now that I'm starting to have a large map size I'm going to start doing quests, and I'd like to do a quest with guards escorting the player (like in Exiled Kingdom), only I don't know how to make NPCs move.
If anyone knows how to do it :)

**Comment by dorkster** [comment]
Assuming you NPC has a `run` animation and has their speed stat set to something greater than 0, all you need is a dialog section with the `take_a_party` property:

```
[stats]
stat=hp,100
speed=6

[dialog]
topic=Recruit guard
take_a_party=true
```

**Reply by BuzzLEquerre** [reply]
Thank you I configured my NPC only he doesn't attack enemies, and seems not to be able to die do I have any solutions ?
my NPC : 

name=Knight, test

portrait=images/portraits/npc_Tuld_mine_knight_enter.png

# animation info
animations=animations/enemies/orc_regular.txt


talker=true

#first dialog
[dialog]
topic=Talk

[stats]
stat=hp,240
speed=4
turn_delay=400ms
chance_pursue=10

power=melee,164,20
passive_powers=185

stat=accuracy,100
stat=avoidance,25
stat=poise,26

stat=dmg_melee_min,50
stat=dmg_melee_max,110
cooldown=1550ms

stat=absorb_min,2
stat=absorb_max,4s


[dialog]
topic=Recruit guard
take_a_party=true

**Reply by dorkster** [reply]
The reason it isn't attacking is because you need to include `threat_range` in your stats block. This is the distance from an enemy the NPC needs to be before engaging in combat.

There shouldn't be anything preventing your ally from dying. Although, I *did* notice a bug in that you can still talk to your ally after they die!

**Reply by BuzzLEquerre** [reply]
It's done !! thk you ! And it is possible to give an inventory or something to treat my NPC during the quest ?

**Reply by dorkster** [reply]
Unfortunately, no. They function the same as enemies, so they don't have their own inventories.

**Reply by BuzzLEquerre** [reply]
ok it's not a big deal I can already do a lot of fun quests with it ! 
And it is possible to indicate a point A and have it be the NPC who takes the player like that : 
![Capture d'écran 2024-11-05 215307](https://github.com/user-attachments/assets/0b9ebba0-d5aa-4cab-af2d-7cc8ddace382)

**Reply by dorkster** [reply]
I'm not quite sure what you're asking. It sounds like you want the NPC to join the player and then leave the party once they reach a certain point. If so, that's not really possible at the moment, since the party status of an NPC can only be controlled by player initiated dialog.

**Reply by BuzzLEquerre** [reply]
Oh sorry my english is not good xd, I meant that once the player starts the quest, the NPC goes to a specific place ex : he goes to a city by showing the way to the player

**Reply by dorkster** [reply]
Thanks for clarifying. But again, that isn't possible at the moment. Allies can only follow the player or move to attack.

**Reply by BuzzLEquerre** [reply]
Okok, thks ! :)
